File: src/AdventOfCode2020/Inputs.kt
package AdventOfCode2020

import java.io.File

const val BASE_PATH = "src/AdventOfCode2020"

val day1 = File("$BASE_PATH/day1.in").readLines().map { it.toInt() }
val day2 = File("$BASE_PATH/day2.in").readLines()
val day3 = File("$BASE_PATH/day3.in").readLines()
val day4 = File("$BASE_PATH/day4.in").readText().split(Regex("\r?\n\r?\n")).map { it.replace(Regex("\r?\n"), " ") }
val day5 = File("$BASE_PATH/day5.in").readLines()
val day6 = File("$BASE_PATH/day6.in").readText().split(Regex("\r?\n\r?\n")).map { it.split(Regex("\r?\n")) }
val day7 = File("$BASE_PATH/day7.in").readLines()
val day8 = File("$BASE_PATH/day8.in").readLines()
val day9 = File("$BASE_PATH/day9.in").readLines()
val day10 = File("$BASE_PATH/day10.in").readLines()
val day11 = File("$BASE_PATH/day11.in").readLines()
val day12 = File("$BASE_PATH/day12.in").readLines()
val day13 = File("$BASE_PATH/day13.in").readLines()

val day15 = "16,12,1,0,15,7,11".split(',').map { it.toInt() } // Short so file io is excessive
val day16 = File("$BASE_PATH/day16.in").readText().split(Regex("\r?\n\r?\n"))





val day22 = File("$BASE_PATH/day22.in").readText().split(Regex("\r?\n\r?\n"))

File: src/AdventOfCode2023/inputs/inputs.go
package inputs

import (
	"os"
	"regexp"
	"strings"
)

const BasePath = "src/AdventOfCode2023"

func getInput(p string) string {
	dat, _ := os.ReadFile(BasePath + "/" + p)
	return string(dat)
}

var Day1 = regexp.MustCompile("\r?\n").Split(getInput("day1.in"), -1)
var Day2 = regexp.MustCompile("\r?\n").Split(getInput("day2.in"), -1)
var Day3 = regexp.MustCompile("\r?\n").Split(getInput("day3.in"), -1)
var Day4 = regexp.MustCompile("\r?\n").Split(getInput("day4.in"), -1)
var Day5 = regexp.MustCompile("\r?\n\r?\n\\w+-to-\\w+ map:\r?\n").Split(getInput("day5.in"), -1)
var Day6 = regexp.MustCompile("\r?\n").Split(getInput("day6.in"), -1)

var Day8 = regexp.MustCompile("\r?\n\r?\n").Split(getInput("day8.in"), -1)
var Day9 = regexp.MustCompile("\r?\n").Split(getInput("day9.in"), -1)
var Day10 = regexp.MustCompile("\r?\n").Split(getInput("day10.in"), -1)
var Day11 = regexp.MustCompile("\r?\n").Split(getInput("day11.in"), -1)

var Day13 = regexp.MustCompile("\r?\n\r?\n").Split(getInput("day13.in"), -1)
var Day14 = regexp.MustCompile("\r?\n").Split(getInput("day14.in"), -1)
var Day15 = strings.Split(getInput("day15.in"), ",")
var Day16 = regexp.MustCompile("\r?\n").Split(getInput("day16.in"), -1)

var Day18 = regexp.MustCompile("\r?\n").Split(getInput("day18.in"), -1)
var Day19 = regexp.MustCompile("\r?\n\r?\n").Split(getInput("day19.in"), -1)

var Day21 = regexp.MustCompile("\r?\n").Split(getInput("day21.in"), -1)
